### PR TITLE
Feature/all customers merchant

### PR DESCRIPTION
**Pull request by Paul-Knapp**
Create method to show all customers for a merchant and test files.

### CODE REVIEW

**Review by devklynch — APPROVED** [review]
Good use of joins!

File: app/models/customer.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Customer < ApplicationRecord
-  has_many :invoices
-
+  has_many :invoices, dependent: :destroy
+  # has_many :merchants, through: :invoices
   validates :first_name, :last_name, presence: true
 
   def self.show_all_customers(merchant_id)
-    Customer.joins(:invoices).where("invoices.merchant_id = ?", merchant_id ).distinct
+    joins(:invoices).where("invoices.merchant_id = ?", merchant_id ).distinct
```
Comment (TDManning) [inline app/models/customer.rb:7]: great job figuring out joins prior to the lesson!

File: config/routes.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@
   put "/api/v1/items/:id", to: "api/v1/items#update"
 
 #customers below
-  get "/api/v1/merchants/:id/customers", to: "api/v1/customers#show"
+  get "/api/v1/merchants/:id/customers", to: "api/v1/merchant_customers#index"
```
Comment (TDManning) [inline config/routes.rb:31]: Good catch/change to the #index

**Comment by TDManning** [comment]
Nice job taking on a more challenging end point!

**Pull request merged**